### PR TITLE
Ongoing OIDC: adds Claims Scopes API, SDK and CLI support

### PR DESCRIPTION
**Pull request by jimlambrt**

### CODE REVIEW

**Review by jefferai — APPROVED** [review]
Looks great!

File: internal/auth/oidc/claims_scope.go [hunk 1]
```diff
@@ -40,7 +44,7 @@ func (cs *ClaimsScope) validate(caller errors.Op) error {
 	if cs.Scope == "" {
 		return errors.New(errors.InvalidParameter, caller, "missing claims scope")
 	}
-	if cs.Scope == "openid" {
+	if cs.Scope == DefaultClaimsScope {
```
[the remark this answers is not in the document]
Comment (jimlambrt) [reply internal/auth/oidc/claims_scope.go:47]: I think, I'd rather call it out as an error and just be consistent that these are "optional" vs the defaults.

[the remark this answers is not in the document]
Comment (jefferai) [reply internal/auth/oidc/claims_scope.go:47]: 👍

File: internal/servers/controller/handlers/authmethods/authmethod_service_test.go [hunk 2]
```diff
@@ -570,6 +570,10 @@ func TestCreate(t *testing.T) {
 						lv, _ := structpb.NewList([]interface{}{"foo", "bar"})
 						return structpb.NewListValue(lv)
 					}(),
+					"claims_scopes": func() *structpb.Value {
```
Comment (talanknight) [inline internal/servers/controller/handlers/authmethods/authmethod_service_test.go:573]: Consider also adding test coverage for the DefaultClaimsScope validation check.

**Review by talanknight — APPROVED** [review]

File: internal/servers/controller/handlers/authmethods/authmethod_service_test.go (hunk 2 above)
Comment (jimlambrt) [reply internal/servers/controller/handlers/authmethods/authmethod_service_test.go:573]: ty!

File: internal/proto/local/controller/api/resources/authmethods/v1/auth_method.proto [hunk 3]
```diff
@@ -12,162 +12,175 @@ import "controller/custom_options/v1/options.proto";
 
 // AuthMethod contains all fields related to an Auth Method resource
 message AuthMethod {
```
[the remark this answers is not in the document]
Comment (jimlambrt) [reply internal/proto/local/controller/api/resources/authmethods/v1/auth_method.proto:14]: Hum... happily bringing this up with the team, so we can perhaps standardize on a linter for protos.  ty!

**Comment by jimlambrt** [comment]
@stellarsquall @randallmorey -> just making sure you saw these changes.

**Pull request merged**